--- tboot-1.8.1/README.orig	2013-06-27 19:09:50.712171198 -0400
+++ tboot-1.8.1/README	2013-06-27 19:09:06.629173035 -0400
@@ -9,9 +9,6 @@ and verified launch of an OS kernel/VMM.
 This version of tboot supports Intel (both retail and Software Development
 Platforms (SDPs)) and OEM systems that are Intel TXT-capable.
 
-This version of tboot only supports both the Xen virtual machine monitor
-(versions >= 3.4) and Linux kernel versions >= 2.6.33.
-
 The mercurial source code repository for this project is located at:
 http://hg.code.sf.net/p/tboot/code.
 
