From 1523c012d394fef36cf5db6cb9b12b5e4e1a8701 Mon Sep 17 00:00:00 2001
From: Ning Sun <ning.sun@intel.com>
Date: Tue, 3 Nov 2015 14:57:03 -0800
Subject: Make a minor change to tboot.c

to avoid a potential security vulnerability.

Signed-off-by: Safayet Ahmed <Safayet.Ahmed@ge.com>
Signed-off-by: Ning Sun <ning.sun@intel.com>
---
 tboot/common/tboot.c |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/tboot/common/tboot.c b/tboot/common/tboot.c
index 08c5e8a..22b9ec0 100644
--- a/tboot/common/tboot.c
+++ b/tboot/common/tboot.c
@@ -175,14 +175,14 @@ static void post_launch(void)
      * verify e820 table and adjust it to protect our memory regions
      */
 
-    /* ensure all modules are in RAM */
-    if ( !verify_modules(g_ldr_ctx) )
-        apply_policy(TB_ERR_POST_LAUNCH_VERIFICATION);
-
     /* marked mem regions used by TXT (heap, SINIT, etc.) as E820_RESERVED */
     err = txt_protect_mem_regions();
     apply_policy(err);
 
+    /* ensure all modules are in RAM */
+    if ( !verify_modules(g_ldr_ctx) )
+        apply_policy(TB_ERR_POST_LAUNCH_VERIFICATION);
+
     /* verify that tboot is in valid RAM (i.e. E820_RAM) */
     base = (uint64_t)TBOOT_BASE_ADDR;
     size = (uint64_t)((unsigned long)&_end - base);
-- 
1.7.1

