From 93e55839aef188cfffc19cf143b63e13c61fe5af Mon Sep 17 00:00:00 2001
From: Ning Sun <ning.sun@intel.com>
Date: Wed, 21 Oct 2015 10:45:01 -0700
Subject: Added an ACPI_RSDP structure g_rsdp in tboot

to avoid potential memory overwritten issue on TPM 2.0 UEFI platforms.

Signed-off-by Ning Sun  <ning.sun@intel.com>
---
 tboot/txt/txt.c |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/tboot/txt/txt.c b/tboot/txt/txt.c
index 321ffd8..b33ef2b 100644
--- a/tboot/txt/txt.c
+++ b/tboot/txt/txt.c
@@ -69,6 +69,7 @@
 /* counter timeout for waiting for all APs to enter wait-for-sipi */
 #define AP_WFS_TIMEOUT     0x01000000
 
+__data struct acpi_rsdp g_rsdp;
 extern char _start[];             /* start of module */
 extern char _end[];               /* end of module */
 extern char _mle_start[];         /* start of text section */
@@ -82,7 +83,6 @@ extern struct mutex ap_lock;
 
 /* MLE/kernel shared data page (in boot.S) */
 extern tboot_shared_t _tboot_shared;
-
 extern void apply_policy(tb_error_t error);
 extern void cpu_wakeup(uint32_t cpuid, uint32_t sipi_vec);
 extern void print_event(const tpm12_pcr_event_t *evt);
@@ -533,7 +533,8 @@ static txt_heap_t *init_txt_heap(void *ptab_base, acm_hdr_t *sinit,
                 os_sinit_data->efi_rsdt_ptr = (uint64_t) rsdp->rsdp1.rsdt;
             } else {
                 /* rsdp */
-                os_sinit_data->efi_rsdt_ptr = (uint64_t)((uint32_t) rsdp);
+                memcpy((void *)&g_rsdp, rsdp, sizeof(struct acpi_rsdp));
+                os_sinit_data->efi_rsdt_ptr = (uint64_t)((uint32_t)&g_rsdp);
             }
         } else {
             /* per discussions--if we don't have an ACPI pointer, die */
-- 
1.7.1

