From 71c832249574b4dfca8126e7310ddf51d365c39b Mon Sep 17 00:00:00 2001
From: Ning Sun <ning.sun@intel.com>
Date: Thu, 7 May 2015 14:57:39 -0700
Subject: Disable -fstack-check in CFLAG for compatibility with Gentoo Linux.

Signed-off-by: Jason Zaman <jason@perfinion.com>
Signed-off-by: Ning Sun <ning.sun@intel.com>
---
 tboot/Config.mk |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/tboot/Config.mk b/tboot/Config.mk
index 826b5e9..67f4f68 100644
--- a/tboot/Config.mk
+++ b/tboot/Config.mk
@@ -30,6 +30,7 @@ CFLAGS		+= -msoft-float
 CFLAGS		+= $(call cc-option,$(CC),-nopie,)
 CFLAGS		+= $(call cc-option,$(CC),-fno-stack-protector,)
 CFLAGS		+= $(call cc-option,$(CC),-fno-stack-protector-all,)
+CFLAGS		+= $(call cc-option,$(CC),-fno-stack-check,)
 
 # changeset variable for banner
 CFLAGS		+= -DTBOOT_CHANGESET=\""$(shell ((hg parents --template "{isodate|isodate} {rev}:{node|short}" >/dev/null && hg parents --template "{isodate|isodate} {rev}:{node|short}") || echo "2015-05-08 12:00 -0800 1.8.3") 2>/dev/null)"\"
-- 
1.7.1

